### PR TITLE
VB-2995 Manage prison contact information

### PR DESCRIPTION
**Pull request by hutcheonb-moj**
* Rename prison 'Status' sub-navigation to 'Configuration'
* Add support for managing prison contact details (create, update, delete)

### CODE REVIEW

File: server/routes/prisons/configuration/prisonConfigController.ts [hunk 1]
```diff
@@ -2,17 +2,57 @@ import { RequestHandler } from 'express'
...
+      const information = await this.prisonService.getPrisonConfig(res.locals.user.username)
+      if (information.email === '') {
+        information.email = 'Not set'
+      }
+      if (information.phone === '') {
+        information.phone = 'Not set'
+      }
+      if (information.website === '') {
+        information.website = 'Not set'
+      }
```
Comment (hutcheonb-moj) [inline server/routes/prisons/configuration/prisonConfigController.ts:23]: not happy with this section - but it makes life simpler if doing this in the route level - as attaching data tests when you've got if statements on a value becomes problematic

File: server/routes/prisons/configuration/prisonConfigController.ts [hunk 2]
```diff
@@ -2,17 +2,57 @@ import { RequestHandler } from 'express'
...
+      const information = await this.prisonService.getPrisonConfig(res.locals.user.username)
+
+      let value = ''
+      if (field === 'email') {
+        value = information.email
+      } else if (field === 'phone') {
+        value = information.phone
+      } else if (field === 'website') {
+        value = information.website
+      } else {
```
Comment (hutcheonb-moj) [inline server/routes/prisons/configuration/prisonConfigController.ts:47]: same as above

File: server/data/prisonRegisterApiClient.ts [hunk 3]
```diff
@@ -12,4 +13,14 @@ export default class PrisonRegisterApiClient {
...
   }
+
+  // will need to add prisonId
+  async getPrisonConfig(): Promise<PrisonConfig> {
+    const information = {
+      email: 'HMPPS-prison-visits@hewell.gov.uk',
+      phone: '',
+      website: 'https://www.gov.uk/guidance/hewell-prison',
+    }
+    return information
```
Comment (hutcheonb-moj) [inline server/data/prisonRegisterApiClient.ts:24]: returns an example piece of data

**Review by hutcheonb-moj — COMMENTED** [review]
file name changes etc make this look a bigger PR than it really is

**Review by tpmcgowan — COMMENTED** [review]
Looks good so far - have added some comments and we can discuss further...

File: integration_tests/e2e/prisons/configuration/prisonConfig.cy.ts [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+    cy.task('stubSignIn')
+    cy.task('stubAuthUser')
+    cy.task('stubPrisons')
+    cy.task('stubGetAllPrisons')
+    cy.signIn()
+
+    cy.task('stubGetPrisonContactDetails', { prisonCode })
+  })
+
+  describe('Prison contact details', () => {
```
Comment (tpmcgowan) [inline integration_tests/e2e/prisons/configuration/prisonConfig.cy.ts:24]: No integration tests yet for add/edit contact details. Will do as a separate PR.

File: server/@types/express/index.d.ts [hunk 5]
```diff
@@ -23,6 +23,7 @@ export declare global {
       id: string
       flash(type: 'errors', message: FlashErrorMessage): number
       flash(type: 'formValues', message: Record<string, string | string[]>): number
+      flash(type: 'formValues'): [Record<string, string | string[]>]
```
Comment (tpmcgowan) [inline server/@types/express/index.d.ts:26]: Add typing for retrieving form values as well as setting them

File: server/routes/prisons/configuration/addEditContactDetailsController.ts [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+        req.flash('formValues', req.body)
+        return res.redirect(originalUrl)
+      }
+
+      const { emailAddress, phoneNumber, webAddress } = req.body
+      const contactDetailsAreEmpty = emailAddress === '' && phoneNumber === '' && webAddress === ''
+
+      try {
+        if (contactDetailsAreEmpty) {
+          await this.prisonService.deletePrisonContactDetails(res.locals.user.username, prisonId)
```
Comment (tpmcgowan) [inline server/routes/prisons/configuration/addEditContactDetailsController.ts:93]: If editing and all fields are set to empty then the contact details are deleted. Otherwise we end up with an empty contact details 'container' stored in the prison register because the fields are all optional.

File: server/views/pages/prisons/sessionTemplates/viewSingleSessionTemplate.njk [hunk 7]
```diff
@@ -235,15 +235,13 @@
       </form>
 
       <h2 class="govuk-heading-m">Update this template</h2>
-      <form action="/prisons/{{ sessionTemplate.prisonId }}/session-templates/{{ sessionTemplate.reference }}/edit" data-test="template-edit-form" method="GET" novalidate>
```
Comment (tpmcgowan) [inline server/views/pages/prisons/sessionTemplates/viewSingleSessionTemplate.njk:238]: Unrelated to this PR but spotted that this button was a form. Shouldn't be as it is just a link and using it in a form was causing the CSRF value to be passed around as a URL parameter (because it's a `GET` form)

**Review by hutcheonb-moj — COMMENTED** [review]
LGTM

**Review by psoleckimoj — APPROVED** [review]
LGTM!
